include/aspect/initial_conditions/interface.h

```diff
@@ -93,6 +94,31 @@ namespace aspect
     };
 
 
+    /**
+     * Create a class inheriting from dealii.Function to translate the
+     * interface call to dealii.Function format
+     */
+    template <int dim>
+    class InterfaceFunction : public dealii::Function<dim, double>
+    {
+      public:
+        InterfaceFunction (const Interface<dim> &interface,
+                           const unsigned int selected_component,
+                           const unsigned int n_components);
```

Comment:
I could not work out how to get the correct inherited initial_temperature function to be called, so the point of the class is to ensure that the implementation in the inheriting function is used. The call cannot be done using &InitialConditions::Interface<dim>::initial_temperature. If you could point me to an appropriate way of doing this, I will make use of that approach instead.